### PR TITLE
Create Plottable object and tests.

### PR DESCRIPTION
**Pull request by drhodges**

### CODE REVIEW

File: emperor/support_files/js/model.js [hunk 1]
```diff
@@ -1,14 +1,97 @@
...
+/**
+ *
+ * @name Plottable
+ *
+ * @class Represents a sample and the associated metadata in the ordination
+ * space.
+ *
+ */
+
 /* Represents an object that can be drawn (i.e. a sample or taxa)
```
Comment (josenavas) [inline emperor/support_files/js/model.js]: I think we can remove this comment, as it is outdated. Your comment is more extensive.

File: emperor/support_files/js/model.js [hunk 2]
```diff
@@ -1,14 +1,97 @@
...
+ *
+ * @class Represents a sample and the associated metadata in the ordination
+ * space.
+ *
+ * @param {name} a string indicating the name of the sample.
+ * @param {idx} the index where the object is located in a decomposition model.
+ * @param {metadata} an Array of strings with the metadata values.
+ * @param {coordinates} an Array of floats indicating the position in space
+ *                      where this sample is located.
+ *
```
Comment (josenavas) [inline emperor/support_files/js/model.js]: Remove this extra line.

**Comment by emperor-helper** [comment]
The test build for this pull request can be found here: http://emperor.colorado.edu/pull_394/make_emperor/

File: emperor/support_files/js/model.js [hunk 3]
```diff
@@ -1,14 +1,97 @@
...
+
+  if (this.ci.length !== 0){
+    if (this.ci.length !== this.coordinates.length){
+      throw new Error("The number of confidence intervals doesn't match with"+
+                      " the number of dimensions in the coordinates "+
+                      "attribute. coords: " + this.coordinates.length +
+                      " ci: " + this.ci.length);
+    }
+  }
+}
```
Comment (josenavas) [inline emperor/support_files/js/model.js]: Add semicolons at the end of the functions. Applies to all.

File: tests/javascript_tests/test_plottable.js [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+
+    plot = new Plottable('foo', ['a', 'b', 'c'], [0.2, 0.3, 0.5, 0.2], 1,
+                         [3, 2, 3, 4]);
+
+    equal(plot.name, 'foo', "The name matches!");
+    deepEqual(plot.metadata, ['a', 'b', 'c'], "The metadata matches!");
+    deepEqual(plot.coordinates, [0.2, 0.3, 0.5, 0.2], "The coordinates "+
+              "match!");
+
+    // optional arguments get default values
```
Comment (josenavas) [inline tests/javascript_tests/test_plottable.js]: This comment do not correspond to the code, since it is not testing that they get the default values.

**Comment by josenavas** [comment]
few comments @drhodges thanks for your contribution!

File: emperor/support_files/js/model.js [hunk 5]
```diff
@@ -1,14 +1,97 @@
...
+ * Helper method to convert a Plottable into a string.
+ *
+ * @return a string describing the Plottable object.
+ *
+ */
+Plottable.prototype.toString = function(){
+  var ret = 'Sample: ' + this.name + ' located at: (' +
+            this.coordinates.join(', ') + ') metadata: ' +
+            this.metadata.join(', ');
+
```
Comment (mortonjt) [inline emperor/support_files/js/model.js]: Would we want to wrap `[  ]` around the metadata values?

i.e, replace with
```
var ret = 'Sample: ' + this.name + ' located at: (' +
          this.coordinates.join(', ') + ') metadata: [' +
          this.metadata.join(', ') + ']';

Also, do we want to display `this.ci` and `this.idx` if they are set?
```

**Comment by mortonjt** [comment]
:+1: once @josenavas comments are addressed

**Comment by josenavas** [comment]
:+1: once tests passes

**Pull request merged**